src/persist/benches/end_to_end.rs

```diff
@@ -196,24 +200,39 @@ fn write_test_data(
     let data_size_bytes = data_size_mb * 1024 * 1024;
     let num_records = (data_size_bytes / record_size_bytes) as u64;
 
-    // TODO(aljoscha): it's hard to write straight from a non-materialized Iterator, without
-    // collecting in a Vec
-    let data: Vec<_> = (0..num_records)
-        .map(|i| {
-            (
-                (generate_bytes(key_bytes), generate_bytes(value_bytes)),
-                i as u64,
-                1,
-            )
-        })
-        .collect();
+    // write data in batches
+
+    let step_size = 1000;
```

Comment:
same as before, now some key sizes will have many more batches than others. that + the fact that we don't physically delete batches i think is also part of why the 4b test uses more disk (the more batches you have the more compactions you will do which without physical deletions just makes more batches)

Reply:
yeah, might be interesting to chunk based on the size of data instead of number of records